cassandra/pool.py

```diff
@@ -450,7 +452,19 @@ def _get_connection_for_routing_key(self, routing_key=None):
         shard_id = None
         if not self._session.cluster.shard_aware_options.disable and self.host.sharding_info and routing_key:
             t = self._session.cluster.metadata.token_map.token_class.from_key(routing_key)
-            shard_id = self.host.sharding_info.shard_id_from_token(t.value)
+            
+            if self._session.cluster.check_tablets_enabled() and self._session.cluster._load_balancing_policy._cluster_metadata._tablets is not None and table is not None:
+                if keyspace is None:
+                    keyspace = self._keyspace
+
+                replicas = self._session.cluster._load_balancing_policy._cluster_metadata._tablets.get_replicas_for_key(keyspace, table, t)
+
+                if replicas != []:
+                    shard_id = replicas[0][1]
+                else:
+                    shard_id = self.host.sharding_info.shard_id_from_token(t.value)
+            else:
+                shard_id = self.host.sharding_info.shard_id_from_token(t.value)
```

Comment:
I'd like to avoid repetition of the else clause. After applying my other comments this could look like this:
```python
            shard_id = None
            if self._session.cluster.check_tablets_enabled() and table is not None:
                if keyspace is None:
                    keyspace = self._keyspace

                replicas = self._session.cluster._load_balancing_policy._cluster_metadata._tablets.get_replicas_for_key(keyspace, table, t)

                if replicas != []:
                    shard_id = replicas[0][1]

            if shard_id is None:
                shard_id = self.host.sharding_info.shard_id_from_token(t.value)
```